Doc/library/typing.rst

```diff
@@ -1275,6 +1275,12 @@ These can be used as types in annotations. They all support subscription using
 
    .. versionadded:: 3.11
 
+.. data:: ReadOnly
+
+   See :pep:`705`. Indicates that a :class:`TypedDict` item may not be modified.
```

Comment:
This documentation should be a bit more detailed and not just refer to the PEP, as the CPython documentation should stand on its own. The typing-extensions docs can refer to just the PEP for experimental features, but not CPython.

I would start with copying over the docstring here and see if we need anything else.